js/client-chat.js

```diff
@@ -1427,8 +1427,8 @@
 						// but it's now always applied
 						$messages = this.$chat.find('.chatmessage-' + user);
 						if (!$messages.length) break;
-						$messages.hide().addClass('revealed').find('button').parent().remove();
-						this.$chat.children().last().append(' <button name="toggleMessages" value="' + user + '" class="subtle"><small>(' + $messages.length + ' line' + ($messages.length > 1 ? 's' : '') + ' from ' + user + ' hidden)</small></button>');
+						$messages.slice(-row[3]).hide().addClass('revealed').find('button').parent().remove();
+						this.$chat.children().last().append(' <button name="toggleMessages" value="' + user + '" class="subtle"><small>(' + $messages.slice(-row[3]).length + ' line' + ($messages.slice(-row[3]).length > 1 ? 's' : '') + ' from ' + user + ' hidden)</small></button>');
```

Comment:
You're using `$messages.slice(-row[3])` three times here, you should really cache the result instead of making three new jQuery objects.